cs/battle.go

```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"math"
 	"sort"
+	"math/rand"
```

Comment:
Can you use the `b.rules.random` instead of math.rand? It makes unit testing mockable and it also allows for seeding the random number generator for a game.